test/cache.c

```
@@ -13800,7 +13800,7 @@ check_check_evictions_enabled_err(unsigned paged)
 
         (cache_ptr->resize_ctl).incr_mode = H5C_incr__threshold;
 
-        result = H5C_get_evictions_enabled(cache_ptr, false);
+        result = H5C_get_evictions_enabled(cache_ptr, NULL);
```

It looks like `H5C_get_evictions_enabled` expects that non-`NULL` will be passed in here, so you may have to simply pass a temporary variable set to false.

---

I looked at the other code using `H5C_get_evictions_enabled` and it sends in `NULL` so I assumed it was ok. `H5C_get_evictions_enabled` also checks if it's `NULL` or not.

---

@jhendersonHDF I looked at the test case again. It's expected to fail if I read it correctly.
```
        result = H5C_get_evictions_enabled(cache_ptr, NULL);

        if (result == SUCCEED) {

            pass         = false;
            failure_mssg = "H5C_get_evictions_enabled succeeded() 2.\n";
        }
```

---

Yep that particular test case is just checking to see that a failure happens when NULL is passed in. I'm assuming the other two cases fixed in this PR were probably a copy-paste issue from the nearby `H5C_set_evictions_enabled` call.

---

The CI will pass in the meantime since it looks like the other two cases that were fixed were intending for the test to fail anyway, but definitely want to make sure it's failing for the intended reason rather than because `NULL` is passed.

---

Ah, ok, I see. I'll update.

---

Not as far as I'm currently aware, but adding `-VV` to ctest should give you some output to see what's going on. `ctest -VV -R H5TEST-accum .` to make life easier.

---

Ok, there seems to be some problem loading the shared libraries in some tests:
```
    Start 21: H5TEST-accum

21: Test command: /home/ted/git/hdf5/build/bin/accum
21: Working Directory: /home/ted/git/hdf5/build/test/H5TEST
21: Environment variables:
21:  srcdir=/home/ted/git/hdf5/build/test/H5TEST
21: Test timeout computed to be: 1200
21: Testing the metadata accumulator
21: Testing simple write/read to/from metadata accumulator                 PASSED
21: Testing simple write/read to/from before metadata accumulator          PASSED
21: Testing simple write/read to/from after metadata accumulator           PASSED
21: Testing overlapping write to metadata accumulator                      PASSED
21: Testing overlapping write to partially clean metadata accumulator      PASSED
21: Testing overlapping write to accumulator larger then accum_size        PASSED
21: Testing non-overlapping write to accumulator larger then accum_size    PASSED
21: Testing accumulator adjustments after append/prepend of data           PASSED
21: Testing reading data from both accumulator and disk                    PASSED
21: Testing simple freeing metadata accumulator                            PASSED
21: Testing large metadata I/O operations                                  PASSED
21: Testing random writes to accumulator                                   PASSED
21: Testing SWMR write of large metadata: with latest format              accum_swmr_reader: error while loading shared libraries: libhdf5_test.so.1000: cannot open shared object file: No such file or directory
21: *FAILED*
21:    at /home/ted/git/hdf5/test/accum.c:2248 in test_swmr_write_big()...
21: child process exited abnormally
21: Testing SWMR write of large metadata: with non-latest-format          accum_swmr_reader: error while loading shared libraries: libhdf5_test.so.1000: cannot open shared object file: No such file or directory
21: *FAILED*
21:    at /home/ted/git/hdf5/test/accum.c:2248 in test_swmr_write_big()...
21: child process exited abnormally
21: *** TESTS FAILED ***
2/3 Test #21: H5TEST-accum .....................***Failed    0.15 sec
```

---

Interesting. Yes your changes almost certainly aren't the cause of those failures, though if `H5TEST-cache` is still failing it would be good to know why.

---

I'm actually getting two
```
76: check_check_evictions_enabled_err: failure_mssg = "H5C_get_evictions_enabled succeeded() 1.
76: check_check_evictions_enabled_err: failure_mssg = "H5C_get_evictions_enabled succeeded() 2.
```
With the dummy `bool` - so... change back to `NULL` I guess?

---

Looks like you'll have to initialize at least this variable to `false` since the logic is checking to make sure that the function both failed and didn't set the variable to `true`, which it must be accidentally getting initialized as here.

---

Hmm, based on the logic here, I'm not sure whether the test was fully finished in the first place. It looks like it was intending to check that the function fails in both cases and doesn't set the variable to `true` in this first one, but if both cases aren't actually failing, I'm not sure what the real intention was.

---

`H5C_get_evictions_enabled` never reads the value of the pointed at `bool` though - and it fails whatever I set it to.

---

As long as `NULL` isn't passed for either of the parameters, `H5C_get_evictions_enabled` should never fail, so it looks like this may be another copy-paste issue. Going back through the git history, I believe both of those calls are actually supposed to be calls to `H5C_set_evictions_enabled` instead, where passing `false` for both would actually make sense again.

---

Well, that works too and the test passes since `H5C_set_evictions_enabled(cache_ptr, false);` fails in both cases.

---

Thanks! Based off the git history and the comment at the start of the test, it definitely looks like the intention was for these to be calls to `H5C_set_evictions_enabled` and something got missed in translation many years ago.

---

Funny what adventures trying to help people debug issues at Stackoverflow can lead to :-)
I came to this repo via https://stackoverflow.com/questions/79385144/cannot-write-a-variable-length-string-attribute-to-hdf5-file

---

Funnily enough, we've had questions very recently about a similar situation. In that particular case I believe the program needs to pass `&value.c_str()` rather than `value.c_str()`. Variable-length strings are a bit of a special case in HDF5. It's potentially a bit misleading due to the way the array is declared, but https://github.com/HDFGroup/hdf5/blob/develop/HDF5Examples/C/H5T/h5ex_t_vlstring.c is an example of variable-length strings from the perspective of the C API. The sort of it is essentially that the library expects `char *` for strings and `char **` for variable-length strings.

---

Ahhhh... wow! Yes, I noticed that the inner function did `H5MM_memcpy(&s, _vl, sizeof(char *));` so that makes sense. I'll try it out. Thanks!